src/components/molecules/UserProfilePicture/UserProfilePicture.tsx

```diff
@@ -16,154 +17,176 @@ import {
-    },
-    [user.id]
+  // TODO: I believe we only need this state to support the imageErrorHandler functionality.. can we just remove it?
+  const [pictureUrl, setPictureUrl] = useState(
+    avatarUrl(user.id, user.anonMode, user.pictureUrl, miniAvatars)
   );
+  useEffect(() => {
+    setPictureUrl(
+      avatarUrl(user.id, user.anonMode, user.pictureUrl, miniAvatars)
+    );
+  }, [miniAvatars, user.anonMode, user.id, user.pictureUrl]);
 
-  return useMemo(() => {
-    return (
-      <S.Container style={{ ...containerStyle }}>
-        {/* Hidden image, used to handle error if image is not loaded */}
-        <img
-          src={pictureUrl}
-          onError={(event) => imageErrorHandler(event)}
-          hidden
-          style={{ display: "none" }}
-          alt={user.anonMode ? DEFAULT_PARTY_NAME : user.partyName}
-        />
-        <S.Avatar
-          onClick={() => openUserProfileModal(user)}
-          className={avatarClassName}
-          backgroundImage={pictureUrl}
-          style={{ ...avatarStyle }}
-        />
+  const openProfileModal = useCallback(() => openUserProfileModal(user), [
+    openUserProfileModal,
+    user,
+  ]);
+
+  // // TODO: extract this logic into a generic custom component + improve it?
+  // const imageErrorHandler = useCallback(
+  //   (event: React.SyntheticEvent<HTMLImageElement, Event>) => {
+  //     const randomAvatar = randomAvatarUrl(user.id);
+  //     setPictureUrl(randomAvatar);
+  //
+  //     // TODO: this isn't very reacty.. is it even needed..?
+  //     // event.currentTarget.onerror = null;
+  //     // event.currentTarget.src = randomAvatar;
+  //   },
+  //   [user.id]
+  // );
+
+  const containerClasses = classNames(
+    "UserProfilePicture",
+    `UserProfilePicture--reaction-${reactionPosition}`
+  );
+
+  const avatarClasses = classNames(
+    "UserProfilePicture__avatar",
+    avatarClassName
+  );
 
+  const avatarStyles = useMemo(
+    () => ({
+      backgroundImage: `url(${pictureUrl})`,
+      ...avatarStyle,
+    }),
+    [avatarStyle, pictureUrl]
+  );
+
+  const userDisplayName = user.anonMode ? DEFAULT_PARTY_NAME : user.partyName;
+
+  return (
+    <div className={containerClasses} style={containerStyle}>
+      {/* TODO: extract this logic into a generic custom component + improve it? Or just remove it entirely? */}
+      {/* Hidden image, used to handle error if image is not loaded */}
+      {/*<img*/}
+      {/*  src={pictureUrl}*/}
+      {/*  onError={(event) => imageErrorHandler(event)}*/}
+      {/*  hidden*/}
+      {/*  style={{ display: "none" }}*/}
+      {/*  alt={user.anonMode ? DEFAULT_PARTY_NAME : user.partyName}*/}
+      {/*/>*/}
+
+      {/* TODO: can we use src/components/atoms/UserAvatar/UserAvatar.tsx here? Should we? */}
+      <div
+        role="img"
+        aria-label={`${userDisplayName}'s avatar`}
+        className={avatarClasses}
+        style={avatarStyles}
+        onClick={openProfileModal}
+      />
```

Comment:
[TODO] can we use `src/components/atoms/UserAvatar/UserAvatar.tsx` here? Should we?

Reply:
Calling this OOS, converted it to an `@debt` for future

Reply:
See @mike-lvov's comments about this in https://github.com/sparkletown/sparkle/pull/1323#discussion_r617811235

Marking this thread resolved.